python/pyarrow/tests/test_dataset.py

```diff
@@ -3095,10 +3104,30 @@ def test_write_dataset_use_threads(tempdir):
         pa.schema([("part", pa.string())]), flavor="hive")
 
     target1 = tempdir / 'partitioned1'
+    paths_written = []
+
+    def file_visitor(written_file):
+        print(f'Visiting {written_file.path}')
```

Comment:
We should probably remove this print statement at the end before merging?

Reply:
Oops.  Removed.